sdk/core/core-lro/test/abort.test.ts

```diff
@@ -0,0 +1,252 @@
+import assert from "assert";
+import { delay, SimpleTokenCredential, WebResource, HttpHeaders } from "@azure/core-http";
+import { TestClient } from "./utils/testClient";
+import { AbortController } from "@azure/abort-controller";
+import { PollerStoppedError } from "../src";
+
+const testHttpHeaders: HttpHeaders = new HttpHeaders();
+const testHttpRequest: WebResource = new WebResource();
+const basicResponseStructure = {
+  headers: testHttpHeaders,
+  parsedBody: {},
+  request: testHttpRequest,
+  status: 200
+};
+const initialResponse = {
+  ...basicResponseStructure,
+  parsedBody: {
+    started: true
+  }
+};
+const doFinalResponse = {
+  ...basicResponseStructure,
+  parsedBody: {
+    doFinalResponse: true
+  }
+};
+const finalResponse = {
+  ...basicResponseStructure,
+  parsedBody: {
+    finished: true
+  }
+};
+
+describe("Long Running Operations - custom client", function() {
+  it("should support an abort signal sent through the constructor", async function() {
+    const client = new TestClient(new SimpleTokenCredential("my-test-token"));
+    client.setResponses([
+      initialResponse,
+      ...Array(20).fill(basicResponseStructure),
+      doFinalResponse,
+      finalResponse
+    ]);
+
+    const abortController = new AbortController();
+    const poller = await client.startLRO({
+      requestOptions: {
+        abortSignal: abortController.signal
+      }
+    });
+
+    // Testing subscriptions to the poll errors
+    let pollError: Error | undefined;
+    poller.done().catch((e) => {
+      pollError = e;
+    });
+
+    await poller.poll();
+    assert.ok(poller.getState().started);
+    await delay(100);
+    assert.equal(client.totalSentRequests, 10);
+    abortController.abort();
+    await delay(50);
+
+    assert.equal(pollError!.message, "The request was aborted");
+    assert.equal(client.totalSentRequests, 10);
+    poller.stop();
+  });
+
+  it("should support an abort signal sent through the constructor (manual poller)", async function() {
+    const client = new TestClient(new SimpleTokenCredential("my-test-token"));
+    client.setResponses([
+      initialResponse,
+      ...Array(20).fill(basicResponseStructure),
+      doFinalResponse,
+      finalResponse
+    ]);
+
+    const abortController = new AbortController();
+    const poller = await client.startLRO({
+      manual: true,
+      requestOptions: {
+        abortSignal: abortController.signal
+      }
+    });
+
+    // Testing subscriptions to the poll errors
+    let pollError: Error | undefined;
+    poller.done().catch((e) => {
+      pollError = e;
+    });
+
+    await poller.poll(); // Manual polling
+    assert.equal(client.totalSentRequests, 1);
+
+    abortController.abort();
+    await poller.poll(); // Manual polling
+    assert.equal(client.totalSentRequests, 1);
+    assert.ok(poller.isDone());
+    assert.equal(pollError!.message, "The request was aborted");
```

Comment:
An implementation note from Brian- the AbortSignal passed to the async op that created the Poller should cause the Promise to be stopped, rather than the operation to be cancelled, so in this case we should see a PollerStoppedError from the done() promise.